crates/sui-core/src/node_sync/node_follower.rs

```diff
@@ -534,6 +597,52 @@ where
             error: e.to_string(),
         })?
     }
+
+    pub fn sync_checkpoint(
+        &self,
+        checkpoint_contents: &CheckpointContents,
+    ) -> impl Stream<Item = SuiResult> {
+        let futures: FuturesOrdered<_> = checkpoint_contents
+            .iter()
+            .map(|digests| {
+                let (tx, rx) = oneshot::channel();
+                let msg = DigestsMessage::new_for_ckpt(digests, tx);
+                Self::send_msg_with_tx(self.sender.clone(), msg, rx)
+            })
+            .collect();
+        futures
+    }
+
+    pub fn handle_execution_request(
+        &self,
+        digests: impl Iterator<Item = TransactionDigest>,
+    ) -> impl Stream<Item = SuiResult> {
+        let futures: FuturesOrdered<_> = digests
+            .map(|digest| {
+                let (tx, rx) = oneshot::channel();
+                let msg = DigestsMessage::new_for_exec_driver(&digest, tx);
+                Self::send_msg_with_tx(self.sender.clone(), msg, rx)
+            })
+            .collect();
+        futures
+    }
+}
+
+#[async_trait]
+impl<A> DigestHandler<A> for NodeSyncHandle
+where
+    A: AuthorityAPI + Send + Sync + 'static + Clone,
+{
+    async fn handle_digest(&self, follower: &Follower<A>, digests: ExecutionDigests) -> SuiResult {
```

Comment:
Here again, I would send blocks of transactions from the follower on follower block boundaries.